src/charm.py

```diff
@@ -1,104 +1,395 @@
+        """Return the workload version.
+
+        Checks for the presence of /etc/workload-version file
+        and if present, returns the contents of that file. If
+        the file is not present, an empty string is returned.
+
+        Returns:
+            string: A human-readable string representing the version of the workload
+        """
+        if self._container.exists(path=WORKLOAD_VERSION_FILE_NAME):
+            version_file_content = self._container.pull(path=WORKLOAD_VERSION_FILE_NAME).read()
+            return version_file_content
+        return ""
+
+
+def _render_config_file(
+    *,
+    gnb_name: str,
+    cu_f1_interface_name: str,
+    cu_f1_ip_address: str,
+    cu_f1_port: int,
+    du_f1_port: int,
+    cu_n2_interface_name: str,
+    cu_n2_ip_address: str,
+    cu_n3_interface_name: str,
+    cu_n3_ip_address: str,
+    amf_external_address: str,
+    mcc: str,
+    mnc: str,
+    sst: int,
+    tac: int,
+) -> str:
+    """Render CU config file based on parameters.
+
+    Args:
+        gnb_name: The name of the gNodeB
+        cu_f1_interface_name: Name of the network interface used for F1 traffic
+        cu_f1_ip_address: IPv4 address of the network interface used for F1 traffic
+        cu_f1_port: Number of the port used by the CU for F1 traffic
+        du_f1_port: Number of the port used by the DU for F1 traffic
+        cu_n2_interface_name: Name of the network interface used for N2 traffic
+        cu_n2_ip_address: IPv4 address of the network interface used for N2 traffic
+        cu_n3_interface_name: Name of the network interface used for N3 traffic
+        cu_n3_ip_address: IPv4 address of the network interface used for N3 traffic
+        amf_external_address: AMF hostname
+        mcc: Mobile Country Code
+        mnc: Mobile Network Code
+        sst: Slice Selection Type
+        tac: Tracking Area Code
+
+    Returns:
+        str: Rendered CU configuration file
+    """
+    jinja2_env = Environment(loader=FileSystemLoader("src/templates"))
+    template = jinja2_env.get_template("cu.conf.j2")
+    return template.render(
+        gnb_name=gnb_name,
+        cu_f1_interface_name=cu_f1_interface_name,
+        cu_f1_ip_address=cu_f1_ip_address,
+        cu_f1_port=cu_f1_port,
+        du_f1_port=du_f1_port,
+        cu_n2_interface_name=cu_n2_interface_name,
+        cu_n2_ip_address=cu_n2_ip_address,
+        cu_n3_interface_name=cu_n3_interface_name,
+        cu_n3_ip_address=cu_n3_ip_address,
+        amf_external_address=amf_external_address,
+        mcc=mcc,
+        mnc=mnc,
+        sst=sst,
+        tac=tac,
+    )
+
+
+def _get_pod_ip() -> str:
+    """Return the pod IP using juju client.
+
+    Returns:
+        str: The pod IP.
+    """
+    ip_address = check_output(["unit-get", "private-address"])
+    if not ip_address:
+        raise OAIRANCUError("Pod IP address not available")
+    return str(IPv4Address(ip_address.decode().strip()))
+
```

Comment:
What will be the charm state if an  OAIRANCUError is raised ? 
We are not checking  availability of  pod id in collect_status event hadler method.

Reply:
Good point.
It's not updated and the respective check is added.